M7 add new post form and comment form.

ram1117 opened the pull request:
Hi in this PR, I have implemented the following. Please review and provide feedback.

-Get current user from ApplicationController.
-Add new post form to Main page.
-Add comment form to each post page.
-Add like functionality to each post. 


https://github.com/ram1117/blogapp/assets/116718155/fdef0bb8-b400-4948-a3dd-4126aaed4906

ram1117 commented on app/controllers/posts_controller.rb:
```diff
@@ -5,6 +5,26 @@ def index
+      flash[:alert] = 'error creating post'
+    end
+  end
+
```
@CollinsTatang  Hi, i need some clarification on creating a `new` action for posts. I am using the new post form as a partial in `users/index` page and not serving the form in a new html page. wouldn't `create` action be sufficient for this.

CollinsTatang replied:
Hi @ram1117 
It could be sufficient, but according to the project requirement you are expected to create a separate controller action for create and new method.

ram1117 commented on app/controllers/comments_controller.rb:
```diff
@@ -0,0 +1,18 @@
+class CommentsController < ApplicationController
```
@CollinsTatang  Hi, i need some clarification on creating a `new` action for comments too. I am using the new comment form as a partial in `posts/index` page and not serving the form in a new html page. wouldn't `create` action be sufficient for this.

CollinsTatang replied:
Hi @ram1117
It could be sufficient, but according to the project requirement, you are expected to create a separate controller action for create and new method.

ram1117 commented:
-Added `new` action in posts and comments controller.

![image](https://github.com/ram1117/blogapp/assets/116718155/5fc0fc47-ac4c-4e02-ad67-5a1ef6a2cdf8)
![image](https://github.com/ram1117/blogapp/assets/116718155/8ed85286-61ba-4575-b766-29d0df5b79aa)

-Added private methods in posts and comments controller to secure form inputs

![image](https://github.com/ram1117/blogapp/assets/116718155/a7bdf1f7-8221-41a6-9549-be64bbade149)
![image](https://github.com/ram1117/blogapp/assets/116718155/a77d61c4-2838-4a3f-8afd-62cae7312cbc)

-Added `new` route to posts and comments.
![image](https://github.com/ram1117/blogapp/assets/116718155/77017007-2d7b-4d37-8cc6-66a4723bc314)

sumairq reviewed (approved):
### Hi @ram1117 , 

<img src="https://media4.giphy.com/media/BPJmthQ3YRwD6QqcVD/giphy.gif"/>

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
